flows/update_flow_run_state_via_client.py

```diff
@@ -0,0 +1,40 @@
+# /// script
+# dependencies = ["prefect"]
+# ///
+
+"""
+Demonstrates using the client to post-hoc update flow run state using the client.
+"""
+
+from uuid import UUID
+
+from prefect import flow, get_client
+from prefect.client.schemas import State, StateType
+
+
+def fail_anyway(flow_run_id: UUID):
+    with get_client(sync_client=True) as client:
+        flow_run = client.read_flow_run(flow_run_id)
+        client.set_flow_run_state(
+            flow_run_id=flow_run.id,
+            state=State(
+                type=StateType.FAILED,
+                message=f"It's not you {flow_run.parameters['name']}, it's me!",
+            ),
+            force=True,
+        )
+        updated_flow_run = client.read_flow_run(flow_run_id)
+        assert (
+            updated_flow_run.state and updated_flow_run.state.type == StateType.FAILED
+        ), "Flow run state was not updated to failed"
+
+
+@flow(log_prints=True)
+def hello(name: str = "Marvin"):
+    print(f"Hello, {name}!")
+
+
+if __name__ == "__main__":
+    state = hello(return_state=True)
+    assert state.state_details.flow_run_id is not None, "Flow run ID is None"
+    fail_anyway(state.state_details.flow_run_id)
```

Comment:
i made the choice to use `state_details` since this is public and not going anywhere and doesn't require the user to alter their flow source code to get something from `FlowRunContext` or `runtime`